### ISSUE TITLE
Problem with DistributedSampler

### ISSUE THREAD

**Issue by ivan-chai** [Jingkang50/OpenOOD#181]
Hello!

DistributedSampler needs to manually set an epoch counter. See the official documentation for a code snippet:
https://pytorch.org/docs/stable/data.html#torch.utils.data.distributed.DistributedSampler

Otherwise shuffling will not work as expected. The issue potentially affects OE training on ImageNet200, as it uses 2 GPUs.

**Comment by ivan-chai** [comment]
Unfortunately I don't have results with the fixed sampler.

**Comment by zjysteven** [comment]
We have fixed this in #223. We also reran a potentially affected method (ConfBranch) on ImageNet200. The results below show that the effect is negligible.

|             |       reported numbers (near-/far-OOD)     | numbers after rerunning (near-/far-OOD) |
|-------------|:---------------------------:|:-----------------------------------------------------------:|
| AUROC       | 79.10 ± 0.24 / 90.43 ± 0.18 |                 79.09 ± 0.06 / 90.43 ± 0.13                 |
| AUPR        | 82.11 ± 0.30 / 88.67 ± 0.27 |                 81.97 ± 0.03 / 88.60 ± 0.13                 |
| FPR@95      | 61.44 ± 0.34 / 34.75 ± 0.63 |                 61.16 ± 0.69 / 33.72 ± 0.31                 |
| ID Accuracy |         85.92 ± 0.07        |                         86.08 ± 0.07                        |
